src/arraystring.rs

```diff
@@ -31,20 +36,20 @@ pub struct ArrayString<const N: usize> {
 
 impl<const N: usize> ArrayString<N>
 where
-    Self: sealed::ValidCapacity,
+    Self: ValidCapacity,
 {
     /// Creates new empty string.
     ///
     /// ```rust
     /// # use arraystring::prelude::*;
-    /// # let _ = env_logger::try_init();
+    /// # #[cfg(not(miri))] let _ = env_logger::try_init();
     /// let string = ArrayString::<23>::new();
     /// assert!(string.is_empty());
     /// ```
     #[inline]
     pub const fn new() -> Self {
         Self {
-            array: [0; N],
+            array: [0; N], // unsafe { MaybeUninit::uninit().assume_init() }, < miri says UB even if not really, lots faster
```

Comment:
I'm not sure you can assume_init a value that hasn't been touched, as it's "freeze".

Reply:
on the small strings it doesn't change anything, 63 and 255 you gain respectively 2 and 20ns or something like that, however i doubt the average user will use those, a normal string is better at that point. zeroing out allows llvm to apply a few optimizations where in UB it would have to assume any possible value.

Reply:
Yeah, let's leave everything initialized, removes a lot of headaches.